gems/pending/postgres_ha_admin/failover_databases.rb

```diff
@@ -0,0 +1,78 @@
+require 'util/postgres_dsn_parser'
+require 'postgres_ha_admin/postgres_ha_logger'
+require 'pg'
+
+module PostgresHaAdmin
+  class FailoverDatabases
+    include PostgresHaAdmin
+
+    attr_reader :yml_file, :connection_hash, :log, :connection_hash
+
+    def initialize(config_dir, log_dir, connection_params_hash)
+      @yml_file = Pathname.new(config_dir).join(DB_YML_FILE)
+      @log = PostgresHaLogger.new(log_dir)
+      @connection_hash = connection_params_hash
+    end
+
+    def refresh_databases_list(connection = nil)
+      query_repmgr(connection)
+    end
+
+    def all_databases(connection = nil)
```

Comment:
I think this class is supposed to be the class that you are describing (a thin wrapper around the yml file containing the database info) and making the file we read from something we can inject is a good candidate for refactoring, but I think we were trying to just move the source in this PR and make those kind of changes in a follow up.